libslide/src/bignum/add.rs

```diff
@@ -0,0 +1,129 @@
+#![allow(clippy::suspicious_arithmetic_impl)]
+use crate::bignum::Bignum;
+use std::mem;
+use std::ops;
+
+impl ops::Add for Bignum {
+    type Output = Bignum;
+
+    fn add(self, rhs: Bignum) -> Bignum {
+        let is_neg: bool = rhs.is_neg && self.is_neg;
+
+        // this assumes the signs are both positive for now
+        let mut carry: u8 = 0;
+        let mut lhs_size: usize = self.dec.len();
+        let mut rhs_size: usize = rhs.dec.len();
+        let mut lhs_vec: Vec<u8>;
+        let mut rhs_vec: Vec<u8>;
+        let mut res: u8;
+
+        // 1. Handle Decimal
+
+        let mut res_dec: Vec<u8>;
+
+        // make lhs smaller vector
+        // lhs_size = self, rhs_self = rhs
+        if lhs_size > rhs_size {
+            mem::swap(&mut lhs_size, &mut rhs_size);
+            lhs_vec = rhs.dec;
+            rhs_vec = self.dec;
+        } else {
+            lhs_vec = self.dec;
+            rhs_vec = rhs.dec;
+        }
+        res_dec = rhs_vec.clone(); 
+
+        for i in (0..lhs_size).rev() {
+            res = lhs_vec[i] + rhs_vec[i] + carry;
+            res_dec[i] = res % 10;
+            carry = (res > 9) as u8
+        };
+
+        // 2. Handle Integer
+
+        lhs_size = self.int.len();
+        rhs_size = rhs.int.len();
+
+        let mut res_int: Vec<u8>;
+
+        // make lhs smaller vector
+        if lhs_size > rhs_size {
+            mem::swap(&mut lhs_size, &mut rhs_size);
+            lhs_vec = rhs.int;
+            rhs_vec = self.int;
+        } else {
+            lhs_vec = self.int;
+            rhs_vec = rhs.int;
+        }
+        res_int = rhs_vec.clone();
+
+        // computer intial addition
+        for i in 0..lhs_size {
+            res = rhs_vec[i] + lhs_vec[i] + carry;
+            res_int[i] = res % 10;
+            carry = (res > 9) as u8;
```

Comment:
you can even get rid of res:

```suggestion
            res_int[i] += lhs_vec[i] + carry;
            carry = (res_int[i] > 9) as u8;
            res_int[i] %= 10;
```